pkg/dao/repository.go

```diff
@@ -0,0 +1,30 @@
+package dao
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/content-services/content-sources-backend/pkg/db"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/labstack/echo/v4"
+)
+
+type repositoryDaoImpl struct {
+}
+
+func GetRepositoryDao() repositoryDaoImpl {
+	return repositoryDaoImpl{}
+}
+
+func (r repositoryDaoImpl) Create(c echo.Context) error {
+
+	newRepository := models.RepositoryConfiguration{}
+
+	if err := c.Bind(&newRepository); err != nil {
+		return fmt.Errorf("bind error: %w", err)
+	}
```

Comment:
I'm glad you started on a dao layer!  I dont' think the dao layer should bind parameters at all.  (and the layer that does should bind to the an api structure, not a db model IMO).   I sorta see the layers like this:

Handler/Api Layer -> Transforms user input into Api Layer objects (RepositoryInfo for example).   Then transforms those into db model objects (RepositoryConfiguration).  Passes these to the DAO layer.
DAO Layer -> Takes the db layer objects and does creates/updates, returns db layer objects for queries

The dependency graph would look kinda like this:

![image](https://user-images.githubusercontent.com/395077/170549233-19a043bc-7285-4597-87c3-f8b4c50acc54.png)


I had originally thought that the DAO layer could convert between the Api structures and the db models, but then we'd end up with a dep graph like this:

![image](https://user-images.githubusercontent.com/395077/170549338-9da60953-e09b-4805-85bc-57973aa3cbdf.png)

which would be a dependency cycle.

Reply:
So i think the flow would go:   Params -> Handler binds to RepositoryInfo  -> passes to DAO layer, which converts to RepositoryConfiguration (and down the road AuthenticationConfiguration)  -> Saves it to the db

Reply:
and conversely within the list api, the flow would be:

Handler  calls into dao layer, dao layer loads RepositoryConfiguration models -> dao converts to RepositoryInfo structs, returns to handler -> handler renders to json. 

I'm kinda thinking out loud here, so please let me know if this isn't making sense or you have a differing opinion :)